src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -36,14 +55,62 @@ public void setup() {
     testUser.setPassword("1234");
     testUser.setGender(Gender.OTHER);
 
-    // when -> any object is being save in the userRepository -> return the dummy
+    //other user for matching etc
+    otherUser = new User();
+    otherUser.setId(1L);
+    otherUser.setName("otherUser");
+    otherUser.setUsername("testOtherUsername");
+    otherUser.setPassword("1234");
+    otherUser.setGender(Gender.OTHER);
+
+    //give; white card for testing
+    testWhiteCard = new WhiteCard();
+    testWhiteCard.setId(1L);
+    testWhiteCard.setText("CardText");
+    testWhiteCards = new ArrayList<>();
+    testWhiteCards.add(testWhiteCard);
+
+    testGame = new Game();
+    testBlackCard = new BlackCard();
+    testBlackCard.setText("GapText");
+    testBlackCard.setId(1L);
+    otherBlackCard = new BlackCard();
+    otherBlackCard.setId(33L);
+    otherBlackCard.setText("some Text");
+
+    testGame.setId(1L);
+    testGame.setUserId(1L);
+    testGame.setBlackCard(testBlackCard);
+    testGame.setCreationTime(new Date());
+    testGame.setGameStatus(GameStatus.ACTIVE);
+    userBlackCards = new UserBlackCards();
+    userBlackCards.setBlackCards(new ArrayList<>(Collections.singleton(testBlackCard)));
+    // when -> any object is being saved in the userRepository -> return the dummy
     // testUser
     Mockito.when(userRepository.save(Mockito.any())).thenReturn(testUser);
   }
 
+  @Test
+  void getUsers_empty(){
+      List<User> emptyUserList = new ArrayList<>();
+      assertEquals(emptyUserList, userService.getUsers());
+  }
+
+    @Test
+    void getUsers_not_empty(){
+        User user = userService.createUser(testUser);
+        List<User> checkList = Arrays.asList(userService.createUser(testUser));
```

Comment:
Why do you recreate that user here? (already did in l. 101)

Reply:
@JoMuelle could you please answer as it is your code..

Reply:
I think this test was there before, not sure, why it is green though. Fixed now